.github/workflows/testing.yml

```diff
@@ -33,7 +33,9 @@ jobs:
           cache: npm
 
       - name: Install latest npm
-        run: npm install --global npm@latest
+        run: npm install --global npm@8.3
+        # TODO: npm@8.4 is broken on Windows. See https://github.com/npm/cli/issues/4341
+        # run: npm install --global npm@latest
```

Comment:
[ask] Why is this diff produced? Can you rebase the main branch?

Reply:
@ybiquitous Sorry.
I rebased the main branch again, but it doesn't seem to have solved the problem.
Is there something wrong with the way I am doing this?
The rebase is done in the jetbrains IDE, not in the CLI, and the following commands are executed internally.

```
16:45:02.124: [stylelint] git -c credential.helper= -c core.quotepath=false -c log.showSignature=false -c core.commentChar= rebase upstream/main
error: Failed to merge in the changes.
hint: Use 'git am --show-current-patch' to see the failed patch
16:45:33.493: [stylelint] git -c credential.helper= -c core.quotepath=false -c log.showSignature=false add --ignore-errors -A -f -- lib/rules/custom-property-pattern/index.js
16:45:33.690: [stylelint] git -c credential.helper= -c core.quotepath=false -c log.showSignature=false -c core.commentChar= rebase --continue
16:48:55.535: [stylelint] git -c credential.helper= -c core.quotepath=false -c log.showSignature=false push --progress --porcelain origin refs/heads/fix-custom-property-pattern:fix-custom-property-pattern
To github.com:kawaguchi1102/stylelint.git
error: failed to push some refs to 'git@github.com:kawaguchi1102/stylelint.git'
hint: Updates were rejected because the tip of your current branch is behind
!	refs/heads/fix-custom-property-pattern:refs/heads/fix-custom-property-pattern	[rejected] (non-fast-forward)
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
Done
16:49:02.532: [stylelint] git -c credential.helper= -c core.quotepath=false -c log.showSignature=false fetch origin --recurse-submodules=no --progress --prune
16:49:05.626: [stylelint] git -c credential.helper= -c core.quotepath=false -c log.showSignature=false rebase origin/fix-custom-property-pattern
First, rewinding head to replay your work on top of it...
Applying: Fix incorrect column and escaped colons for `selector-pseudo-element-colon-notation` rule (#5879)
Applying: Update CHANGELOG.md
Applying: Fix false negatives for custom properties within var() in custom-property-pattern (#3655)
Using index info to reconstruct a base tree...
M	lib/rules/custom-property-pattern/__tests__/index.js
Falling back to patching base and 3-way merge...
No changes -- Patch already applied.
16:49:05.893: [stylelint] git -c credential.helper= -c core.quotepath=false -c log.showSignature=false push --progress --porcelain origin refs/heads/fix-custom-property-pattern:fix-custom-property-pattern
Enumerating objects: 23, done.
Counting objects:   4% (1/23)
...
Counting objects: 100% (23/23)
Counting objects: 100% (23/23), done.
Delta compression using up to 4 threads
Compressing objects:  11% (1/9)
...
Compressing objects: 100% (9/9)
Compressing objects: 100% (9/9), done.
Writing objects:   7% (1/13)
...
Writing objects: 100% (13/13)
Writing objects: 100% (13/13), 1.82 KiB | 1.82 MiB/s, done.
Total 13 (delta 9), reused 4 (delta 3)
remote: Resolving deltas:   0% (0/9)        
...
remote: Resolving deltas: 100% (9/9)        
remote: Resolving deltas: 100% (9/9), completed with 8 local objects.        
To github.com:kawaguchi1102/stylelint.git
 	refs/heads/fix-custom-property-pattern:refs/heads/fix-custom-property-pattern	accfc917..4af18ae0
Done
```

Reply:
When I try rebasing the main branch on your fork repository, it works. Here are the steps:

```console
$ git clone https://github.com/kawaguchi1102/stylelint.git

$ cd stylelint

$ git checkout fix-custom-property-pattern

$ git rebase main

# Resolve conflicts in lib/rules/custom-property-pattern/index.js by my editor...

$ git add lib/rules/custom-property-pattern/index.js

$ git rebase --continue
```

<details>
<summary><code>git diff main fix-custom-property-pattern</code></summary>

```diff
diff --git a/lib/rules/custom-property-pattern/__tests__/index.js b/lib/rules/custom-property-pattern/__tests__/index.js
index 7dd52973..c6481136 100644
--- a/lib/rules/custom-property-pattern/__tests__/index.js
+++ b/lib/rules/custom-property-pattern/__tests__/index.js
@@ -13,16 +13,59 @@ testRule({
 		{
 			code: ':root { --boo-foo-bar: 0; }',
 		},
+		{
+			code: ':root { --foo-color: #f00; } a { color: var(--foo-color); }',
+		},
+		{
+			code: ':root { --foo-sub-color: #f00; } a { color: var(--foo-color, var(--foo-sub-color)); }',
+		},
 	],
 
 	reject: [
 		{
 			code: ':root { --boo-bar: 0; }',
 			message: messages.expected(/foo-.+/),
+			line: 1,
+			column: 9,
 		},
 		{
 			code: ':root { --foo-: 0; }',
 			message: messages.expected(/foo-.+/),
+			line: 1,
+			column: 9,
+		},
+		{
+			code: ':root { --foo-color: #f00; } a { color: var(--boo-color); }',
+			message: messages.expected(/foo-.+/),
+			line: 1,
+			column: 41,
+		},
+		{
+			code: ':root { --foo-sub-color: #f00; } a { color: var(--foo-color, var(--boo-sub-color)); }',
+			message: messages.expected(/foo-.+/),
+			line: 1,
+			column: 62,
+		},
+		{
+			code: ':root { --foo-color: #f00; } a { color: VAR(--boo-color)); }',
+			message: messages.expected(/foo-.+/),
+			line: 1,
+			column: 41,
+		},
+		{
+			code: ':root { --foo-color: #f00; } a { color: var(--boo-color), var(--boo-sub-color)); }',
+			warnings: [
+				{ message: messages.expected(/foo-.+/), line: 1, column: 41 },
+				{ message: messages.expected(/foo-.+/), line: 1, column: 59 },
+			],
+		},
+		{
+			code: ':root { --foo-color: #f00; } a { color: var(--boo-color); }',
+			message: messages.expected(/foo-.+/),
+		},
+		{
+			code: ':root { --foo-sub-color: #f00; } a { color: var(--foo-color, var(--boo-sub-color)); }',
+			message: messages.expected(/foo-.+/),
 		},
 	],
 });
@@ -44,10 +87,14 @@ testRule({
 		{
 			code: ':root { --boo-bar: 0; }',
 			message: messages.expected('foo-.+'),
+			line: 1,
+			column: 9,
 		},
 		{
 			code: ':root { --foo-: 0; }',
 			message: messages.expected('foo-.+'),
+			line: 1,
+			column: 9,
 		},
 	],
 });
@@ -69,14 +116,20 @@ testRule({
 		{
 			code: ':root { --boo-Foo-bar: 0; }',
 			message: messages.expected(/^[A-Z][a-z]+-[a-z][a-zA-Z]+$/),
+			line: 1,
+			column: 9,
 		},
 		{
 			code: ':root { --foo-bar: 0; }',
 			message: messages.expected(/^[A-Z][a-z]+-[a-z][a-zA-Z]+$/),
+			line: 1,
+			column: 9,
 		},
 		{
 			code: ':root { --Foo-Bar: 0; }',
 			message: messages.expected(/^[A-Z][a-z]+-[a-z][a-zA-Z]+$/),
+			line: 1,
+			column: 9,
 		},
 	],
 });
diff --git a/lib/rules/custom-property-pattern/index.js b/lib/rules/custom-property-pattern/index.js
index b5e51add..1b813631 100644
--- a/lib/rules/custom-property-pattern/index.js
+++ b/lib/rules/custom-property-pattern/index.js
@@ -1,10 +1,13 @@
 'use strict';
 
+const valueParser = require('postcss-value-parser');
 const isCustomProperty = require('../../utils/isCustomProperty');
 const report = require('../../utils/report');
 const ruleMessages = require('../../utils/ruleMessages');
 const validateOptions = require('../../utils/validateOptions');
+const declarationValueIndex = require('../../utils/declarationValueIndex');
 const { isRegExp, isString } = require('../../utils/validateTypes');
+const { isValueFunction } = require('../../utils/typeGuards');
 
 const ruleName = 'custom-property-pattern';
 
@@ -31,7 +34,21 @@ const rule = (primary) => {
 		const regexpPattern = isString(primary) ? new RegExp(primary) : primary;
 
 		root.walkDecls((decl) => {
-			const prop = decl.prop;
+			const { prop, value } = decl;
+
+			const parsedValue = valueParser(value);
+
+			parsedValue.walk((node) => {
+				if (!isValueFunction(node)) return;
+
+				if (node.value !== 'var') return;
+
+				const { nodes, sourceIndex } = node;
+
+				if (regexpPattern.test(nodes[0].value.slice(2))) return;
+
+				complain(messages.expected(primary), declarationValueIndex(decl) + sourceIndex, decl);
+			});
 
 			if (!isCustomProperty(prop)) {
 				return;
@@ -41,13 +58,23 @@ const rule = (primary) => {
 				return;
 			}
 
+			complain(messages.expected(primary), 0, decl);
+		});
+
+		/**
+		 * @param {string} message
+		 * @param {number} index
+		 * @param {import('postcss').Declaration} decl
+		 */
+		function complain(message, index, decl) {
 			report({
-				message: messages.expected(primary),
-				node: decl,
 				result,
 				ruleName,
+				message: messages.expected(primary),
+				node: decl,
+				index,
 			});
-		});
+		}
 	};
 };
 
```

</details>

Reply:
@ybiquitous Thanks for the advice.
I was able to rebase it successfully.